ios/RNBranch.m

```diff
@@ -403,8 +403,17 @@ - (BranchUniversalObject *)findUniversalObjectWithIdent:(NSString *)ident reject
 #pragma mark setIdentity
 RCT_EXPORT_METHOD(
                   setIdentity:(NSString *)identity
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(__unused RCTPromiseRejectBlock)reject
                   ) {
-    [self.class.branch setIdentity:identity];
+    [self.class.branch setIdentity: identity withCallback:^(NSDictionary *params, NSError *error) {
+        if (!error) {
+            resolve([self.class.branch getFirstReferringParams]);
```

Comment:
@nsingh-branch Don't we have to do the same to RNBranchModule.java https://github.com/BranchMetrics/react-native-branch-deep-linking-attribution/blob/master/android/src/main/java/io/branch/rnbranch/RNBranchModule.java#L442 and have the promise.resolve or promise.reject?